CONTRIBUTING.md

```diff
@@ -128,6 +128,86 @@ Two other things to keep in mind:
 - If you add code that should be tested, add tests.
 - If you change the public API, [update the documentation](#api-reference).
 
+### Debugging Environment
+
+Due to the way that python and rust interoperate, debugging the Rust side of development from Python calls can be difficult. For this reason, we have provided
+a guide and several tools to setup up [Visual Studio Code](https://code.visualstudio.com/) with a debugging environment that makes debugging relatively simple.
+
+#### VS Code Extensions
+
+Install The [CodeLLDB](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) and the [Task Shell Input](https://marketplace.visualstudio.com/items?itemName=augustocdias.tasks-shell-input) extensions.
+You can install from within a VS Code terminal with the following (use `-r` to run in the current VSCode window):
+
+```shell
+code -r --install-extension vadimcn.vscode-lldb augustocdias.tasks-shell-input
+code -r --install-extension augustocdias.tasks-shell-input
+```
+
+#### Debug Configuration
+
+Copy the following `configurations` and `inputs` components to your `launch.json` file.
+
+<details><summary><code><b>launch.json</b></code></summary>
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Debug Rust",
+            "type": "python",
+            "request": "launch",
+            "program": "${workspaceFolder}/py-polars/debug/launch.py",
+            "args": [
+                "${file}",
+            ],
+            "console": "internalConsole",
+            "justMyCode": true,
+            "serverReadyAction": {
+                "pattern": "pID = ([0-9]+)",
+                "action": "startDebugging",
+                "name": "Rust LLDB",
+            },
+        },
+        {
+            "name": "Rust LLDB",
+            "type": "lldb",
+            "request": "attach",
+            "program": "${workspaceFolder}/py-polars/.venv/bin/pytho",
+            "pid": "${input:readPIDDirectTestFile}",
+            "stopOnEntry": false,
+            "sourceLanguages": [
+                "rust"
+            ],
+            "presentation": {
+                "hidden": true,
+            },
+        },
+    ],
+    "inputs": [
+        {
+            "id": "readPIDDirectTestFile",
```

Comment:
Couldn't this stuff be removed since you're getting the Pid from `launch.py` instead?
Also if this isn't necessary, also the sections in `Contributing.md` about using this plugin can be removed.

Reply:
So, I made a comment about this in the OP above. I do get the `pID` and I would *like* to remove this dependency, but I currently have no way to send this to the Rust LLDB launch configuration. If you could figure out how to do that, then we could remove the dependency.

As it is, we actually don't need the `Popen` stuff in the python script *at all* right now: it's only used to print a message to the debug console so that the Python debug configuration can detect it and launch the Rust LLDB configuration. We could use the string "hello" and it would work just as fine.

I left it in so that, in the future, maybe we can figure out how to pass it to the Rust LLDB config.

Reply:
@SeanTroyUWO I'm super close to removing the dependency by using `env`/`envFile` in the launch config. The only problem is--env and envFile don't work in CodeLLDB for Linux, although they do in Windows. I created an issue here: https://github.com/vadimcn/codelldb/issues/968.

My other idea use ports/remote debugging, that way we could just use a hardcoded port.

Reply:
Would this help? Could `launch.py` write the Pid to a temp file, which gets read by this plugin? 

https://marketplace.visualstudio.com/items?itemName=rioj7.command-variable#file-content

I played with this when I tried to first get this to work. I couldn't get it to work in one click however. This might have been fixed with your idea of using `launch.py` to call the test python file.

ie:
python debug process starts w/ `launch.py` -> Pid of said debug process found by `launch.py` -> `launch.py` writes this to a temp file somewhere in the polars directory that gets erased by `make clean` and is `.gitignored` -> Command Variable extension reads from file -> Rust debug process starts

Reply:
Maybe, although that adds a new dependency after removing the old one. What I have in place is to write the pID to `/debug/launch.env` and have the LLDB launch read that file, but the env vars are failing. It works fine on Windows though!

Reply:
At this point, I think everything works: we can stop on both Python and Rust breakpoints now. We should keep an eye out to remove this extra dependency in the future, hopefully when the environmental variable issue gets resolved in CodeLLDB. For now, I think we should see if others are willing to accept this.

Thanks so much for the help on this @SeanTroyUWO!

Reply:
Great, I'm glad. Btw I randomly remembered something, do you have to do that `ptrace` thing because you're using WSL? I had a little note about it in my original set of instructions. For me I had to run `echo 0 | sudo tee /proc/sys/kernel/yama/ptrace_scope` once before using the script.

Reply:
> that `ptrace` thing

Can you clarify? I haven't done anything with `ptrace`.

Reply:
@SeanTroyUWO I had another idea, which may be a bad one: simply write the `pID` directly into `launch.json` prior to calling the Rust connector. I'm going to test this out.

Reply:
It works! This means we don't need the extension or that entire `"inputs"` section in the `launch.json` file.

Reply:
Just pushed a new commit with the shell extensions requirement removed. Let me know if it works for you.